app/controllers/admin/articles_controller.rb

```diff
@@ -0,0 +1,53 @@
+class Admin::ArticlesController < ApplicationController
```

Comment:
Create an admin base application controller class and move the http basic auth into that